Gulpfile.js

```diff
@@ -14,110 +15,9 @@ var lint = require('gulp-tslint');
-      .pipe(ts(project))
-      .pipe(sourcemaps.write())
-      .pipe(gulp.dest('./tmp')); // Write every JS file to tmp/
-  }, function() {
-    return gulp.src('./tmp/**/*.js')
-      .pipe(sourcemaps.init())
-      .pipe(webpack({
-        entry: './tmp/main.js',
-        output: {
-          filename: 'app.js'
-        },
-        node: {
-          fs: 'empty'
-        },
-        module: {
-          loaders: [
-            { test: /\.json/, loader: 'json' },
-          ]
-        }
-      }))
-      .pipe(sourcemaps.write())
-      .pipe(gulp.dest('./dist')); // webpack everything to dist/app.js
-    }, function(cb) {
-      del('tmp').then(function() { cb(); });
-    }, function() {
-      return gulp.src('./app/index.html')
-        .pipe(gulp.dest('./dist'));
-    }));
-    gulp.task('lint:'+name, function() {
-      var tsFiles = gulpFilter(['**/*.ts', '!**/*.d.ts']);
-      return project.src()
-        .pipe(tsFiles)
-        .pipe(lint({configuration: lintConfig, tslint: tslint}))
-        .pipe(lint.report(lintReporter, {emitError: false}));
-    });
-}
-
-var testDest = './test/app';
-function makeTestBuildTarget(name, projectfile) {
-  var project = ts.createProject(projectfile, {typescript: typescript});
-  gulp.task(name+'-build', function() {
-    return project.src()
-      .pipe(sourcemaps.init())
-      .pipe(ts(project))
-      .pipe(sourcemaps.write())
-      .pipe(gulp.dest(testDest));
-  });
-  gulp.task(name+'-instrument', function() {
-    return gulp.src(testDest+'/*.js')
-      .pipe(istanbul({coverageVariable: '__coverage__'}))
-      .pipe(gulp.dest(testDest+'/instrumented')) //instrumented file must differ from original
-  });
-  gulp.task(name, gulp.series(name+'-build', name+'-instrument'));
-  gulp.task('lint:'+name, function() {
-    var tsFiles = gulpFilter(['**/*.ts', '!**/*.d.ts']);
-    return project.src()
-      .pipe(tsFiles)
-      .pipe(lint({configuration: lintConfig, tslint: tslint}))
-      .pipe(lint.report(lintReporter, {emitError: false}));
-  });
-}
-
-makeFrontendBuildTarget('build:app', 'app/tsconfig.json');
-makeTestBuildTarget('build:test:app', 'test/app/tsconfig.json');
-
-gulp.task('lint', gulp.parallel('lint:build:app', 'lint:build:test:app'));
-
-gulp.task('build:test', gulp.parallel('build:test:app'));
-
-gulp.task('build', gulp.parallel('build:app', 'build:test'));
-
-gulp.task('test-run', function() {
-  return gulp.src('test/app/runner.html')
-  .pipe(mochaPhantomJS({
-    reporter: 'spec',
-    phantomjs: {
-      hooks: 'mocha-phantomjs-istanbul',
-      coverageFile: coverageFile
-    }
-  }));
-});
-
-gulp.task('test-coverage-report', function() {
-  return gulp.src(coverageFile)
-    .pipe(istanbulReport({
-      reporters: ['text-summary', 'html']
-    }));
-});
-
-gulp.task('test', gulp.series('test-run', 'test-coverage-report'));
+gulp.task('lint', gulp.parallel('lint:app'));
+gulp.task('build', gulp.parallel('build:app'));
+gulp.task('test', gulp.parallel('test:app'));
```

Comment:
Shouldn't these be the other way around? I'd expect the Gulpfile for app to just say `lint`, `build`, or `test` and the overarching Gulpfile to say `build:app`